Sources/NIOHTTP1/NIOTypedHTTPServerUpgradeHandler.swift

```diff
@@ -154,6 +154,25 @@ public final class NIOTypedHTTPServerUpgradeHandler<UpgradeResult: Sendable>: Ch
         }
     }
 
+    public func userInboundEventTriggered(context: ChannelHandlerContext, event: Any) {
+        switch event {
+        case let evt as ChannelEvent where evt == ChannelEvent.inputClosed:
+            // The remote peer half-closed the channel during the upgrade. Should we close the other side
+            switch self.stateMachine.inputClosed() {
+            case .close:
+                self.upgradeResultPromise.fail(ChannelError.inputClosed)
+                context.close(promise: nil)
```

Comment:
Can we switch these two lines around so we first call close and then fulfil the promise. This should avoid potential bugs where the promise callbacks interact with the channel.

Reply:
done